troubleshooting/known_issues/Istio_Pods_Crashing_due_to_fs_inotify_Limits.md

```diff
@@ -0,0 +1,41 @@
+# Istio Pods Crashing Post Upgrade Due to `fs.inotify` Limits
+
+## Issue Description
+
+After upgrading Istio, nodes may not have booted into the new image with the increased `fs.inotify.max_user_instances` and `fs.inotify.max_user_watches` values.
+When pods restart (such as after a power outage or node reboot), they may fail due to insufficient `inotify` resources, as the limits on the system are too low.
+
+This issue manifests when:
+
+- Pods are unable to create enough `inotify` instances to monitor required files.
+- The system hits the maximum number of file watches, causing crashes or failures in services dependent on file system event monitoring.
+
+This problem can be triggered by events like:
+
+- A node dying and rebooting mid-upgrade.
+- Power outages where pods restart on nodes with old kernel settings.
+
+## Related Issue
+
+- [Istio Issue #35829](https://github.com/istio/istio/issues/35829)
+
+The problem was worked around by following the advice to increase `fs.inotify.max_user_instances` from 128 to 1024 as suggested in the GitHub thread.
+
+## Steps to Reproduce
+
+1. Upgrade Istio without rebooting nodes into an image with updated kernel parameter values.
+2. Trigger a pod restart or node reboot.
+3. Observe pods failing to start due to `inotify-related` errors.
```

Comment:
Could you add a specific example of these errors. That way if customers search for the error in our documentation, they will find this page.

Reply:
I have added the error log for this